app/Http/Controllers/Admin/UserController.php

```diff
@@ -189,6 +208,39 @@ public function update(UserFormRequest $request, User $user)
         return redirect()->route('admin.users.view', $user->id);
     }
 
+    /**
+     * Change a user to OAuth2/Normal on the system.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Pterodactyl\Models\User $user
+     * @return \Illuminate\Http\RedirectResponse
+     *
+     * @throws \Exception
+     * @throws \Throwable
+     */
+    public function put(Request $request, User $user)
+    {
+        if (is_null(env('OAUTH2_CLIENT_ID'))) {
+            return redirect()->route('admin.users.view', $user->id);
+        }
+
+        if (isset($user->getAttributes()['oauth2_id'])) {
+            $user->update(['oauth2_id'  => null]);
```

Comment:
Please do not use the models to modify and save the user, they're difficult to unit test properly. You should look at the UserUpdateService or use the user repository to make quick changes like this.